test-data/unit/check-incremental.test

```diff
@@ -3745,6 +3745,183 @@ class C(A, B):
+[file a/__init__.py]
+[file a/a.py]
+from typing import Optional
+def converter(s:Optional[int]) -> int:
+    ...
+
+import attr
+@attr.s
+class A:
+    x: int = attr.ib(converter=converter)
+
+[builtins fixtures/list.pyi]
+[out1]
+main:2: error: Revealed type is 'def (x: Union[builtins.int, builtins.None]) -> a.a.A'
+[out2]
+main:2: error: Revealed type is 'def (x: Union[builtins.int, builtins.None]) -> a.a.A'
+
+[case testAttrsIncrementalConverterManyStyles]
+from base import Base
+reveal_type(Base)
+from subclass import A, B
+reveal_type(A)
+reveal_type(B)
+from submodule.base import SubBase
+reveal_type(SubBase)
+from submodule.subclass import AA, BB
+reveal_type(AA)
+reveal_type(BB)
+from submodule.subsubclass import SubAA, SubBB
+reveal_type(SubAA)
+reveal_type(SubBB)
+
+[file foo.py]
+from typing import Optional
+def maybe_int(x: Optional[int]) -> int:
+   ...
+[file bar.py]
+from typing import Optional
+def maybe_bool(x: Optional[bool]) -> bool:
+   ...
+[file base.py]
+from typing import Optional
+import attr
+import bar
+from foo import maybe_int
+def maybe_str(x: Optional[str]) -> str:
+   ...
+@attr.s
+class Base:
+    x: int = attr.ib(converter=maybe_int)
+    y: str = attr.ib(converter=maybe_str)
+    z: bool = attr.ib(converter=bar.maybe_bool)
+[file subclass.py]
+from typing import Optional
+import attr
+from base import Base
+@attr.s
+class A(Base): pass
+
+import bar
+from foo import maybe_int
+def maybe_str(x: Optional[str]) -> str:
+   ...
+@attr.s
+class B(Base):
+    xx: int = attr.ib(converter=maybe_int)
+    yy: str = attr.ib(converter=maybe_str)
+    zz: bool = attr.ib(converter=bar.maybe_bool)
+
+[file submodule/__init__.py]
+[file submodule/base.py]
+from typing import Optional
+import attr
+import bar
+from foo import maybe_int
+def maybe_str(x: Optional[str]) -> str:
+   ...
+@attr.s
+class SubBase:
+    x: int = attr.ib(converter=maybe_int)
+    y: str = attr.ib(converter=maybe_str)
+    z: bool = attr.ib(converter=bar.maybe_bool)
+
+[file submodule/subclass.py]
+from typing import Optional
+import attr
+from base import Base
+@attr.s
+class AA(Base): pass
+
+import bar
+from foo import maybe_int
+def maybe_str(x: Optional[str]) -> str:
+   ...
+@attr.s
+class BB(Base):
+    xx: int = attr.ib(converter=maybe_int)
+    yy: str = attr.ib(converter=maybe_str)
+    zz: bool = attr.ib(converter=bar.maybe_bool)
+
+[file submodule/subsubclass.py]
+from typing import Optional
+import attr
+from .base import SubBase
+@attr.s
+class SubAA(SubBase): pass
+
+import bar
+from foo import maybe_int
+def maybe_str(x: Optional[str]) -> str:
+   ...
+@attr.s
+class SubBB(SubBase):
+    xx: int = attr.ib(converter=maybe_int)
+    yy: str = attr.ib(converter=maybe_str)
+    zz: bool = attr.ib(converter=bar.maybe_bool)
+[builtins fixtures/list.pyi]
+[out1]
+main:2: error: Revealed type is 'def (x: Union[builtins.int, builtins.None], y: Union[builtins.str, builtins.None], z: Union[builtins.bool, builtins.None]) -> base.Base'
```

Comment:
This is a bit hard to read. Could you please change this to instead checking types of things by assignments that would cause an error on the _second_ run only? For example:
```
import a
[file a.py]
from b import thing
x: str = thing # OK
[file a.py.2]
from b import thing
x: int = thing # Error
[file b.py]
thing = 'thing'
```

Reply:
Ok. It'll be more like checking class creation though.